sky/packages/sky/lib/sky_tool

```diff
@@ -254,6 +284,359 @@ class StopSky(object):
+                 ' application in the simulator and attaches'
+                 ' the debugger to it.')
+        simulator_parser.add_argument('-p', dest='path', required=True,
+            help='Path to the simulator application.')
+        simulator_parser.add_argument('-t', dest='target', required=False,
+            default='examples/demo_launcher/lib/main.dart',
+            help='Sky server-relative path to the Sky app to run.')
+        simulator_parser.add_argument('-s', dest='server', required=False,
+            default='localhost:8080',
+            help='Sky server address.')
+        simulator_parser.add_argument('--ios_sim_path', dest='ios_sim_path',
+            help='Path to your iOS Simulator executable. '
+                 'Not normally required.')
+
+        subparsers = simulator_parser.add_subparsers()
+        launch_parser = subparsers.add_parser('launch', help='Launch app')
+        launch_parser.set_defaults(func=self.launch_app)
+        install_parser = subparsers.add_parser('install', help='Install app')
+        install_parser.set_defaults(func=self.install_app)
+        debug_parser = subparsers.add_parser('debug', help='Debug app')
+        debug_parser.set_defaults(func=self.debug_app)
+
+    def run(self, args, pids):
+      return args.func(args)
+
+
+class StartListening(object):
+    def add_subparser(self, subparsers):
+        listen_parser = subparsers.add_parser('listen',
+            help=('Listen for changes to files and reload the running app on all connected devices'))
+        listen_parser.set_defaults(func=self.run)
+
+    def run(self, args, pids):
+        cmd = [
+            'which',
+            'fswatch'
+        ]
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        out, err = process.communicate()
+        match = re.search(r'fswatch', out)
+        if match is None:
+            logging.error('"listen" command is only useful if you have installed fswatch.  Run "brew install fswatch" to install it with homebrew.')
+            return
+
+        tempdir = None
+        currdir = None
+        while True:
+            # Watch filesystem for changes
+            cmd = [
+                'fswatch',
+                '-r',
+                '-v',
+                '-1',
+                '.'
+            ]
+            subprocess.check_call(cmd)
+
+            logging.info('Updating running Sky apps...')
+
+            # Restart the app on Android.  Android does not currently restart using skyx files.
+            cmd = [
+                os.path.abspath(__file__),
+                'start',
+                '--poke'
+            ]
+            subprocess.check_call(cmd)
+
+            if not args.local_build:
+                # Currently sending to iOS only works if you are building Sky locally
+                # since we aren't shipping the sky_snapshot binary yet.
+                continue
+
+            if tempdir is None:
+                tempdir = tempfile.mkdtemp()
```

Comment:
As it turns out, the default version of python on Mac is sufficiently old that TemporaryDirectory is not available.  I'll fix this in another commit -- it's getting hairy.